modules/xmpp/recording.js

```diff
@@ -1,19 +1,26 @@
 /* global $, $iq, config, connection, focusMucJid, messageHandler,
    Toolbar, Util, Promise */
 var XMPPEvents = require("../../service/xmpp/XMPPEvents");
+var JitsiRecorderErrors = require("../../JitsiRecorderErrors");
+
 var logger = require("jitsi-meet-logger").getLogger(__filename);
 
 function Recording(type, eventEmitter, connection, focusMucJid, jirecon,
     roomjid) {
     this.eventEmitter = eventEmitter;
     this.connection = connection;
-    this.state = "off";
+    this.state = null;
     this.focusMucJid = focusMucJid;
     this.jirecon = jirecon;
     this.url = null;
     this.type = type;
-    this._isSupported = ((type === Recording.types.JIBRI)
-        || (type === Recording.types.JIRECON && !this.jirecon))? false : true;
+    this._isSupported
+        = ( type === Recording.types.JIRECON && !this.jirecon
+            || type === Recording.types.JIBRI && !this._isServiceAvailable)
+            ? false : true;
+
+    this._isServiceAvailable = false;
+
```

Comment:
checking for type === Recording.types.JIBRI && !this._isServiceAvailable will always be the same as _isServiceAvailable is defined after it.